port/unix/omrsysinfo.c

```diff
@@ -3750,6 +3751,40 @@ isRunningInContainer(struct OMRPortLibrary *portLibrary, BOOLEAN *inContainer)
 	return rc;
 }
 
+static int32_t
+getCgroupMemoryLimit(struct OMRPortLibrary *portLibrary, uint64_t *limit)
+{
+	uint64_t cgroupMemLimit = 0;
+	uint64_t physicalMemLimit = 0;
+	int32_t numItemsToRead = 1; /* memory.limit_in_bytes file contains only one integer value */
+	int32_t rc = 0;
+
+	Trc_PRT_sysinfo_cgroup_get_memlimit_Entry();
+
+	rc = readCgroupSubsystemFile(portLibrary, OMR_CGROUP_SUBSYSTEM_MEMORY, "memory.limit_in_bytes", numItemsToRead, "%lu", &cgroupMemLimit);
+	if (0 != rc) {
+		Trc_PRT_sysinfo_cgroup_get_memlimit_memory_limit_read_failed("memory.limit_in_bytes", rc);
+		goto _end;
+	}
+
+	physicalMemLimit = getPhysicalMemory(portLibrary);
```

Comment:
Do we need to get this data and do the checks below to decide if CGroups are set?  Isn't the above code enough to know that CGroups are set?

Reply:
Problem is if cgroup memory limit is not set, then the value in `memory.limit_in_bytes` file is a very large number (actually close to LONG_MAX value). Therefore, the need to explicitly check the value in file against available physical memory to know if the limit is set or not.

Reply:
sigh... ok the cgroup stuff seems like such a hack on the container side... This change is ok.